components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParRequestBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.par.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.oltu.oauth2.common.utils.OAuthUtils;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.central.log.mgt.utils.LogConstants;
+import org.wso2.carbon.identity.central.log.mgt.utils.LoggerUtils;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth2.AbstractRequestBuilder;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.utils.DiagnosticLog;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.LogConstants.InputKeys.REQUEST_BUILDER;
+
+/**
+ * This builds the PAR request if the incoming request satisfies the PAR conditions.
+ */
+public class ParRequestBuilder implements AbstractRequestBuilder {
+
+    private static final Log log = LogFactory.getLog(ParRequestBuilder.class);
+
+    private static final String REQUEST_BUILDER_NAME = "Pushed authorization request builder";
+
+    @Override
+    public HttpServletRequest buildRequest(HttpServletRequest request) throws IdentityException {
+
+        return new OAuthParRequestWrapper(request);
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        // Requests that separately contain the scope `openid` will not be handled in the PAR flow since they are
```

Comment:
We cant think like that, a oidc request can also be a PAR request right? is there any mention regarding this in the spec? if not we can check on this with the oauth community.

if its up to us to decide then i think we may need to 1st try to resolve the request from the PAR builder and if it doesn't resolve send back the original request then it will again go through the oidc request builder if such a one is configured.